<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <!-- Main dialog widget -->
 <class>AcademicDialog</class>
 <widget class="QDialog" name="AcademicDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <!-- Main vertical layout -->
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <!-- Form layout for input fields -->
    <layout class="QFormLayout" name="formLayout">
     <!-- Title field -->
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Title:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="titleEdit"/>
     </item>
     <!-- Description field -->
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Description:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QTextEdit" name="descriptionEdit"/>
     </item>
     <!-- Date/Time field -->
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Date/Time:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QDateTimeEdit" name="dateTimeEdit"/>
     </item>
     <!-- Duration field (for lectures only) -->
     <item row="3" column="0">
      <widget class="QLabel" name="durationLabel">
       <property name="text">
        <string>Duration (minutes):</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QSpinBox" name="durationSpinBox">
       <property name="maximum">
        <number>480</number>
       </property>
       <property name="singleStep">
        <number>15</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <!-- OK/Cancel buttons -->
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <!-- Signal/slot connections -->
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>AcademicDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui> 